### PR TITLE
Add Datadog Agent & Chef Handler

### PR DESCRIPTION
**Pull request by jeffbyrnes**
* Provides monitoring of nodes in DataDog
* Reports Chef runs to DataDog event stream

@eherot, thoughts on bringing this back globally?

### CODE REVIEW

File: recipes/datadog.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# Copyright (c) 2016 EverTrue, All Rights Reserved.
+
+datadog_creds = data_bag_item('secrets', 'api_keys')['datadog']
+node.set['datadog']['api_key'] = datadog_creds['api_key']
+node.set['datadog']['application_key'] = datadog_creds['application_keys']['chef_agent']
+
+%w(
+  dd-agent
+  dd-handler
+).each do |r|
```
Comment (eherot) [inline recipes/datadog.rb]: IMO this `do; end` block should be a one-liner

**Comment by eherot** [comment]
This PR looks good but is it a thing we still want?

**Comment by jeffbyrnes** [comment]
@eherot we should hold off on this; it’s $15 / host, so we should be circumspect for now with setting this up anywhere.

I think I’ll start by seeing if Niall will let us pay for just Mesos & Kafka hosts (since we need introspection on those), which will be 17 hosts ($255 / mo.), and the go from there. I think Datadog can be a huge tool for us, since they have a StatsD implementation, and charge by the host, not per-metric.

**Comment by eherot** [comment]
I'd concur on just setting it up for Mesos (and possibly Kafka) hosts, especially since those will tend to be bigger and fewer in number.

**Comment by jeffbyrnes** [comment]
Sounds good. I’ll close this for now. Tom found we can get good insight into Kafka, so I’ll work w/ Niall to budget out buying Datadog for a few months to properly try it out.

**Comment by jeffbyrnes** [comment]
Oh, please just leave the branch for now; we may want this later.

**Pull request closed without merging**